src/graphics/gameplay/board.c

```diff
@@ -0,0 +1,81 @@
+#include "board.h"
+
+#include <stdlib.h>
+#include <stdbool.h>
+
+#include "../include/raylib.h"
+#include "../include/raymath.h"
+
+#include "../tmp_structs.h"
+#include "../../game_functions/board_functions.h"
+#include "constants.h"
+
+int handleBoardActionEvents (Board *board, Vector2 mouse_position, double width) {
+    float tile_width = ((float) width) / board->cols;
+
+    int selected_tile_row = (int) (mouse_position.y / tile_width);
+    int selected_tile_col = (int) (mouse_position.x / tile_width);
+
+    int selected_tile_index = calcIndexFromPosition(selected_tile_row,  selected_tile_col, board->cols);
+    bool gameEnd = false;
```

Comment:
Would it not make sense to include the gameEnd bool value in the ``GameplayData`` struct?

Reply:
This is potentially not needed yet; however, I have it here incase we need it. I will put it into `GameplayData` incase we use this data. For now, it can just be an unused variable in that function.